### ISSUE TITLE
Cryptic error message when gateway fails to start as a container

### ISSUE THREAD

**Issue by hash-d** [skupperproject/skupper#715]
# How to reproduce

Create a gateway, remove its configmap and secret, and try to create a new gateway.

```
$ skupper gateway status
Gateway Definition:
╰─ GATEWAY-NAME type:podman version:2.0.0-SNAPSHOT
$ oc delete cm skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328f 
configmap "skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328f" deleted
$ oc delete secret skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328
secret "skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328f" deleted
$ skupper gateway init --type podman
Error: Failed to start gateway as container: exit status 125
$ 
```

# Notes

The scenario above is rather artificial, but easy to reproduce.  That said, skupper today does not have logic that 'undoes' part of an unsuccessful gateway creation.  That might be true for other gateway commands as well.  With that, it should be fairly common for this kind of error to be triggered.

If the user looks at `podman(1)`, they'll find that error 125 means `The error is with podman itself`, which does not help any.  I have met this issue several times during testing.  To figure out the underlying error, I had to use either a debugger or a tracer, which is not a very good prospect for a support perspective.

# Suggestions

In the case of the reproduction scenario above, `podman` was returning a fairly descriptive message.

```
Error: error creating container storage: the container name "GATEWAY-NAME" is already in use by "a7953a633d719c592ad84461959d35b3549fc1ff3e69ecd1252acbc903ef2d76". You have to remove that container to be able to reuse that name.: that name is already in use.
```

The suggestions is to capture `podman`'s output and show it to the user in case of errors.

**Comment by hash-d** [comment]
@ajssmith, I have just tested this and it worked fine on master. Can you confirm this was addressed by https://github.com/skupperproject/skupper/pull/771?

Once you do, I'll test it with 1.0 and close. I just want to make sure the issue and fix are properly tied together.

**Comment by ajssmith** [comment]
@hash-d yes this was addressed by #771 as it detects existing container in use

**Comment by hash-d** [comment]
I tested it against 1.1.1, and it returns a proper answer.

```
  $ skupper status
  Skupper is enabled for namespace "default" in interior mode. It is not connected to any other sites. It has no exposed services.
  The site console url is:  https://10.104.142.16:8080
  The credentials for internal console-auth mode are held in secret: 'skupper-console-users'
  $ skupper version
  client version                 1.1.1
  transport version              quay.io/skupper/skupper-router:2.1.0 (sha256:0e3887c570cf)
  controller version             quay.io/skupper/service-controller:1.1.1 (sha256:f984aa2a0589)
  config-sync version            quay.io/skupper/config-sync:1.1.1 (sha256:f918c9284c70)
  $ skupper gateway status
  No gateway definition found on cluster
  $ skupper gateway init --type podman
  Skupper gateway: 'dhashimo-dhashimo'. Use 'skupper gateway status' to get more information.
  $ skupper gateway status
  Gateway Definition:
  ╰─ dhashimo-dhashimo type:podman version:2.1.0
  $ oc delete cm skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328f
  configmap "skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328f" deleted
  $ oc delete secret skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328
  Error from server (NotFound): secrets "skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328" not found
  $ skupper gateway init --type podman
  Error: Failed to create gateway secret: secrets "skupper-gateway-8c08bc5ba454dd12e0aef053f7655ae60fa8328f" already exists
  $ 
```
